### ISSUE TITLE
Unescaped characters in info.json from e-hentai/exhentai gallery names

### ISSUE THREAD

**Issue by Dystasia** [HDoujinDownloader/HDoujin-Downloader#13]
Double quote and Tab characters used in gallery names from e-hentai/exhentai and/or other sites are not escaped correctly in the info.json files. This results in parsing issues when using software that requires standard JSON format.

From the RFC 4627 section 2.5:

- Backspace to be replaced with \b
- Form feed to be replaced with \f
- Newline to be replaced with \n
- Carriage return to be replaced with \r
- **Tab to be replaced with \t**
- **Double quote to be replaced with \\"**
- Backslash to be replaced with \\\

Example of info.json that's unparseable (notice how even github markdown recognizes that there is a problem by highlighting the issue):
``` json
{
  "manga_info": {
    "title": "(C95) [Fry Dish (Jakko)] "Katte ni" Coconatsu Fan Kanshasai (Hinabita)",
    "original_title": "(C95) [ふらいでぃっしゅ (じゃっこ)] 「勝手に」ここなつファン感謝祭 (ひなビタ♪)",
    "author": [],
    "artist": [
      "Jakko"
    ],
    "circle": [
      "Fry Dish"
    ],
    "scanlator": [],
    "translator": [],
    "publisher": "",
    "description": "",
    "status": "",
    "chapters": "N/A",
    "pages": 25,
    "tags": {
      "female": [
        "Chinese Dress",
        "Hair Buns",
        "Mind Control",
        "Pantyhose",
        "Stockings",
        "Twintails"
      ],
      "male": [
        "Glasses"
      ],
      "misc": [
        "Group"
      ]
    },
    "type": "Doujinshi",
    "language": [
      "Japanese"
    ],
    "released": "",
    "reading_direction": "",
    "characters": [],
    "series": "",
    "parody": [
      "Beatmania",
      "Hinabita"
    ],
    "url": "https://exhentai.org/g/1369492/2484bc5859/"
  }
}
```

The correctly escaped version is: 
``` json
{
  "manga_info": {
    "title": "(C95) [Fry Dish (Jakko)] \"Katte ni\" Coconatsu Fan Kanshasai (Hinabita)",
    "original_title": "(C95) [ふらいでぃっしゅ (じゃっこ)] 「勝手に」ここなつファン感謝祭 (ひなビタ♪)",
    "author": [],
    "artist": [
      "Jakko"
    ],
    "circle": [
      "Fry Dish"
    ],
    "scanlator": [],
    "translator": [],
    "publisher": "",
    "description": "",
    "status": "",
    "chapters": "N/A",
    "pages": 25,
    "tags": {
      "female": [
        "Chinese Dress",
        "Hair Buns",
        "Mind Control",
        "Pantyhose",
        "Stockings",
        "Twintails"
      ],
      "male": [
        "Glasses"
      ],
      "misc": [
        "Group"
      ]
    },
    "type": "Doujinshi",
    "language": [
      "Japanese"
    ],
    "released": "",
    "reading_direction": "",
    "characters": [],
    "series": "",
    "parody": [
      "Beatmania",
      "Hinabita"
    ],
    "url": "https://exhentai.org/g/1369492/2484bc5859/"
  }
}
```
Note: I don't have an example right now of galleries with unscaped tab characters because I have fixed them, but they exists. 

All of these galleries' metadata fail to be imported in software such as HPX and others manga managers.

**Comment by Dystasia** [comment]
@HDoujinDownloader I made a small .NET program to fix the most prominent of these unscaped characters double quotes. This is done after the fact (meaning after info.json get created I iterate over them and fix them), but it would be best to fix it at the source HDoujinDownloader =).

If you want I can give you my snipped code, but I don't know if you could use it inside your context.

Let me know.

**Comment by HDoujinDownloader** [comment]
You're absolutely right. When I first added JSON as a metadata output format many years ago, my implementation was very naïve and didn't fully conform to the spec, as you've noted. Oddly enough, you're the first person to actually mention it, and it's about time it's fixed.

I've now got it fixed on my end, and it should be all good after the next update (which I hope to release shortly).

**Comment by Dystasia** [comment]
Nice to hear that! I am also surprise by this not being reported much earlier. I was wondering if people where even using this feature (info.json). I thought it was super useful and use it every time I can. IMO, anyone's database is only as good as the content used to generate it and the info.json is essential to import galleries into new manga manager databases without having to do complete crawling or matadata refreshes.

**Comment by HDoujinDownloader** [comment]
Totally agree with you-- A library with no means of filtering and organizing content is next to useless.

This should now be resolved with version `v2021.2.7.0`.
